### PR TITLE
Data Explorer: Add polars sorting and row filtering speed up get_data_values, process schema changes to invalidate filters and sort columns

### PR DESCRIPTION
**Pull request by wesm**
Addresses polars support items #3451, #3452, #3761 and did some refactoring for better code reuse. 

I also realized that pydantic validation (and subsequent conversion back to a raw dict for going over the comm interface) of `get_data_values` requests was adding extra overhead that is not so useful in practice, so I have bypassed this validation to make these requests a lot faster. 

### QA Notes

Run same sorting and filter workflows that you would with pandas.

### CODE REVIEW

**Comment by wesm** [comment]
I added the schema state updates work here. I'll put subsequent work (copy-paste and others) in follow up PRs

**Review by seeM — COMMENTED** [review]
A quick read through the code changes looks good.

One issue while testing: if you try to start the kernel from a fresh virtual environment it fails to start because of the `pytz` import in `data_explorer.py` (EDIT: Might be good to add a test for fresh envs at some point):

```
[Python] Traceback (most recent call last):
[Python]   File "/Users/seem/posit/positron/extensions/positron-python/python_files/positron/positron_language_server.py", line 12, in <module>
[Python]     from positron_ipykernel.positron_ipkernel import PositronIPKernelApp
[Python]   File "/Users/seem/posit/positron/extensions/positron-python/python_files/positron/positron_ipykernel/positron_ipkernel.py", line 33, in <module>
[Python]     from .data_explorer import DataExplorerService
[Python]   File "/Users/seem/posit/positron/extensions/positron-python/python_files/positron/positron_ipykernel/data_explorer.py", line 26, in <module>
[Python]     import pytz
[Python] ModuleNotFoundError: No module named 'pytz'
```

On Python 3.11.9, I then installed pandas (v2.2.2) and polars (0.20.31) and tried out scrolling, sorting, and filtering. It all worked well!

File: extensions/positron-python/python_files/positron/positron_ipykernel/variables.py [hunk 1]
```diff
@@ -149,6 +150,8 @@ def handle_msg(
 
         else:
             logger.warning(f"Unhandled request: {request}")
+            if utils.TESTING:
```
[the remark this answers is not in the document]
Comment (wesm) [reply extensions/positron-python/python_files/positron/positron_ipykernel/variables.py:153]: I'll remove this, I was getting frustrated about exceptions in message RPC handlers not being raised (to the point that `--pdb` is useful) in the unit tests and was tinkering with possible solutions to that, but this wasn't it

**Comment by wesm** [comment]
> One issue while testing: if you try to start the kernel from a fresh virtual environment it fails to start because of the pytz import in data_explorer.py

will fix -- I agree that we should capture this case in CI!

**Pull request merged**